torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1579,7 +1594,10 @@ void ProcessGroupNCCL::watchdogHandler() {
               // Set shutdown mode, so the heartbeat monitor thread will not
               // abort process immediately.
               collectiveDebugInfoMode_.store(true);
-              std::vector<uint8_t> vec(1);
+              int rank = globalRank();
+              auto vec = std::vector<uint8_t>(
```

Comment:
confused about the set api.  are we forced to use vector<uint8_t>?  Or why not use a type that matches the type used for rank?